opensrp-chw/src/main/java/org/smartregister/chw/interactor/FpUpcomingServicesInteractor.java

```diff
@@ -2,21 +2,138 @@
+            serviceDueDate = alertRule.getDueDate();
+            serviceOverDueDate = alertRule.getOverDueDate();
+            if (fpMethodUsed.equalsIgnoreCase(FamilyPlanningConstants.DBConstants.FP_INJECTABLE)) {
+                serviceName = fpMethod;
+            } else {
+                serviceName = MessageFormat.format(context.getString(R.string.refill), fpMethod);
+            }
+        } else if (fpMethodUsed.equalsIgnoreCase(FamilyPlanningConstants.DBConstants.FP_IUCD) || fpMethodUsed.equalsIgnoreCase(FamilyPlanningConstants.DBConstants.FP_FEMALE_STERLIZATION)) {
+            if (lastVisit == null) {
+                serviceDueDate = alertRule.getDueDate();
+                serviceOverDueDate = alertRule.getOverDueDate();
+                serviceName = MessageFormat.format(context.getString(R.string.follow_up_one), fpMethod);
+            } else {
+                if (fpMethodUsed.equalsIgnoreCase(FamilyPlanningConstants.DBConstants.FP_IUCD)) {
+                    serviceDueDate = (DateTimeFormat.forPattern("dd-MM-yyyy").parseLocalDate(fp_date).plusMonths(4)).toDate();
+                    serviceOverDueDate = (DateTimeFormat.forPattern("dd-MM-yyyy").parseLocalDate(fp_date).plusMonths(4).plusWeeks(1)).toDate();
+                    serviceName = MessageFormat.format(context.getString(R.string.follow_up_two), fpMethod);
+                } else {
+                    count = FpDao.getCountFpVisits(memberObject.getBaseEntityId(), FamilyPlanningConstants.EventType.FP_FOLLOW_UP_VISIT, fpMethod);
+                    if (count == 2) {
+                        serviceDueDate = (DateTimeFormat.forPattern("dd-MM-yyyy").parseLocalDate(fp_date).plusMonths(1)).toDate();
+                        serviceOverDueDate = (DateTimeFormat.forPattern("dd-MM-yyyy").parseLocalDate(fp_date).plusMonths(1).plusWeeks(2)).toDate();
+                        serviceName = MessageFormat.format(context.getString(R.string.follow_up_three), fpMethod);
+                    } else {
+                        serviceDueDate = (DateTimeFormat.forPattern("dd-MM-yyyy").parseLocalDate(fp_date).plusDays(7)).toDate();
+                        serviceOverDueDate = (DateTimeFormat.forPattern("dd-MM-yyyy").parseLocalDate(fp_date).plusDays(9)).toDate();
+                        serviceName = MessageFormat.format(context.getString(R.string.follow_up_two), fpMethod);
+                    }
+                }
+            }
+        }
+        BaseUpcomingService upcomingService = new BaseUpcomingService();
+        upcomingService.setServiceDate(serviceDueDate);
+        upcomingService.setOverDueDate(serviceOverDueDate);
+        upcomingService.setServiceName(serviceName);
+        serviceList.add(upcomingService);
+
+    }
+
+    private String getTranslatedValue(@Nullable String fpMethod){
```

Comment:
- I'd suggest moving this to a Util since it is already being used in more than one place, and also it'd be easier to add/edit different FP methods strings/translations if they were in one location
- Also, the same fixes applied earlier should also be applied where there's `FpDao.getLatestInjectionVisit()` & `FpDao.getLatestFpVisit()` so that translated `fpMethod` string shouldn't be one of the arguments. 
- Looks like the only place the translated values are used in this class should be when setting the `serviceName`